native/src/Navigator.tsx

```diff
@@ -112,63 +113,61 @@ const Navigator = (): ReactElement | null => {
     )
   }, [])
 
-  const updateInitialRoute = useCallback(
-    (initialRoute: InitialRouteType) =>
-      // Do not override initial route set by opening push notification
-      setInitialRoute(previous => (previous.name === REDIRECT_ROUTE ? previous : initialRoute)),
-    []
-  )
-
   useEffect(() => {
-    const initialize = async () => {
-      const usingHermes = typeof HermesInternal === 'object' && HermesInternal !== null
-
-      if (usingHermes) {
-        log('App is using Hermes: https://reactnative.dev/docs/hermes')
-      }
-
-      const { introShown, storageVersion, errorTracking } = await appSettings.loadSettings()
-
-      if (errorTracking) {
-        initSentry()
-      }
-
-      if (!storageVersion) {
-        await appSettings.setVersion(ASYNC_STORAGE_VERSION)
-      }
-
-      if (storageVersion !== ASYNC_STORAGE_VERSION) {
-        // start a migration routine
-      }
-
-      if (!buildConfig().featureFlags.introSlides && !introShown) {
-        await appSettings.setIntroShown()
-      }
-
-      if (buildConfig().featureFlags.introSlides && !introShown) {
-        updateInitialRoute({
-          name: INTRO_ROUTE,
-        })
-      } else if (cityCode) {
-        updateInitialRoute({
-          name: CATEGORIES_ROUTE,
-        })
-      } else {
-        updateInitialRoute({
-          name: LANDING_ROUTE,
-        })
-      }
-
-      setWaitingForSettings(false)
+    if (!settings) {
+      return
     }
+    const { errorTracking, storageVersion, introShown } = settings
+    const usingHermes = typeof HermesInternal === 'object' && HermesInternal !== null
 
-    initialize().catch(error => setErrorMessage(error.message))
-  }, [cityCode, updateInitialRoute, setErrorMessage])
+    if (usingHermes) {
+      log('App is using Hermes: https://reactnative.dev/docs/hermes')
+    }
 
-  if (errorMessage) {
-    return <Text>{errorMessage}</Text>
+    if (errorTracking) {
+      initSentry()
+    }
+
+    if (!storageVersion) {
+      appSettings.setVersion(ASYNC_STORAGE_VERSION).catch(reportError)
+    }
+
+    if (storageVersion !== ASYNC_STORAGE_VERSION) {
+      // start a migration routine
+    }
```

Comment:
Is this a todo for the future? Then we can maybe add a `TODO` to make it easier to find? Or can it be removed?

Reply:
Its not a todo, more like an instruction if our async storage/app settings schema changes at some point where to put migration routines. But yeah, I think we can remove it.